finagle-http/src/main/scala/com/twitter/finagle/http/codec/HttpServerDispatcher.scala

```diff
@@ -31,6 +31,11 @@ private[finagle] class HttpServerDispatcher(
     service.close()
   }
 
+  private def mayHaveContent(status: Status): Boolean = status match {
```

Comment:
Hi @monkey-mas, I'm new to this code so feel free to let me know if what follows is incorrect!
I don't think that `handle(res: Response)` is called multiple times: it's responsible for writing the message to the wire so if it's called multiple times we're in big trouble. I do think you're right in that its going to be called after all the filters so in its current form, its a problem.
It sounds like the problem is accommodating people sending a 204 with a body. In the case of people sending a 204 _without_ a body, the content-length header gets set to 0. Maybe what we can do is have `handle` look for a 204 response with a `content-length: 0` header, and strip those from the response? That will make 'legit' 204 responses legal by rfc 7230, and give your filter a way to signal that the header should be stripped.

To make this happen, the single `if` statement on 96 (old) might turn into:
```scala
rep.contentLength match {
  case Some(0) if rep.status == Status.NoContent =>
    rep.headers.remove(Fields.ContentLength)

  case Some(_) => // NOOP: don't perturb explicitly set content lengths other than 204 with length 0.  

  case None =>
    rep.contentLength = rep.content.length
}
```
There might be a cleaner way to implement that logic.

The filter would then set the `content-length` headers for Status(304), set it to 0 for Status(204), and strip away the bodies.